src/main/java/knightminer/inspirations/library/util/ReflectionUtil.java

```diff
@@ -0,0 +1,131 @@
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.potion.PotionType;
+import net.minecraftforge.fml.relauncher.ReflectionHelper;
+import net.minecraftforge.fml.relauncher.ReflectionHelper.UnableToFindFieldException;
+import net.minecraftforge.fml.relauncher.ReflectionHelper.UnableToFindMethodException;
+import net.minecraftforge.fml.relauncher.ReflectionHelper.UnableToFindClassException;
+import net.minecraftforge.registries.IRegistryDelegate;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+@SuppressWarnings("unchecked")
+public final class ReflectionUtil {
+	private ReflectionUtil() {}
+
+	private static final Map<String, Field> FIELDS = new HashMap<>();
+	private static final Map<String, Method> METHODS = new HashMap<>();
+	private static final Map<String, Class> CLASS = new HashMap<>();
+
+	@Nullable
+	public static PotionType getMixPredicateInput(@Nonnull Object mixPredicate) {
+		Object val = getFieldValue(getClass("net.minecraft.potion.PotionHelper$MixPredicate"), mixPredicate, "input", "field_185198_a");
+		if(val instanceof IRegistryDelegate) {
+			return ((IRegistryDelegate<PotionType>) val).get();
+		} else { // fallback for older Forge versions
+			return (PotionType) val;
+		}
+	}
+
+	@Nullable
+	public static Ingredient getMixPredicateReagent(@Nonnull Object mixPredicate) {
+		return getFieldValue(getClass("net.minecraft.potion.PotionHelper$MixPredicate"), mixPredicate, "reagent", "field_185199_b");
+	}
+
+	@Nullable
+	public static PotionType getMixPredicateOutput(@Nonnull Object mixPredicate) {
+		Object val = getFieldValue(getClass("net.minecraft.potion.PotionHelper$MixPredicate"), mixPredicate, "output", "field_185200_c");
+		if(val instanceof IRegistryDelegate) {
+			return ((IRegistryDelegate<PotionType>) val).get();
+		} else { // fallback for older Forge versions
+			return (PotionType) val;
+		}
+	}
+
+	@Nullable
+	public static ItemStack invokeGetSilkTouchDrop(@Nonnull Block block, IBlockState state) {
+		ItemStack stack = invokeMethod(Block.class, block, "getSilkTouchDrop", "func_180643_i", new Class[] {IBlockState.class}, state);
+		if(stack != null) {
+			return stack;
+		} else {
+			return ItemStack.EMPTY;
+		}
+	}
+
+	/**
+	 * Looks up a class by its name, caches it for further use and returns it.<br>
+	 * If it can't find the class, it will be logged and <tt>null</tt> is returned.
+	 *
+	 * @param className The class name to be searched for
+	 * @return The class found or <tt>null</tt>, if the class is unavailable
+	 */
+	public static Class getClass(String className) {
+		try {
+			return CLASS.computeIfAbsent(className, key -> ReflectionHelper.getClass(Inspirations.class.getClassLoader(), key));
```

Comment:
Does this need to be the main class, or can it use something like the registry. Reason I ask is because for the most part I keep classes in library from referencing anything outside of library.

Reply:
I usually use the main class, but sure, I can use the registry